src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -704,6 +709,36 @@ void GZBridge::odometryCallback(const gz::msgs::OdometryWithCovariance &odometry
 	pthread_mutex_unlock(&_node_mutex);
 }
 
+void GZBridge::navSatCallback(const gz::msgs::NavSat &nav_sat)
+{
+	if (hrt_absolute_time() == 0) {
+		return;
+	}
+
+	pthread_mutex_lock(&_node_mutex);
+
+	const uint64_t time_us = (nav_sat.header().stamp().sec() * 1000000) + (nav_sat.header().stamp().nsec() / 1000);
+
+	if (time_us > _world_time_us.load()) {
+		updateClock(nav_sat.header().stamp().sec(), nav_sat.header().stamp().nsec());
+	}
+
+	_timestamp_prev = time_us;
+
+	double latitude_deg = nav_sat.latitude_deg();
+	double longitude_deg = nav_sat.longitude_deg();
+	double altitude = nav_sat.altitude();
```

Comment:
As for discussed, for now how about simply publishing `vehicle_global_position_groundtruth` from the NavSat callback?

```suggestion
	
	_gpos_ground_truth_pub.publish(global_position_groundtruth);
```

Reply:
@dagar I have moved the `vehicle_global_position_groundtruth` to the navsat callback and removed the global groundtruth from being assigned in `poseInfoCallback`. This also necessitated adding a check if `_pos_ref` has been initalized as it could otherwise lead to a case where poseInfo is called without `_pos_ref` being initialized (as this is now done in the navsat callback).

Reply:
You can publish everything else that's not available without the pos_ref.

This is what ekf2 does on board. https://github.com/PX4/PX4-Autopilot/blob/f636414ca7ea42239c6d85772fe0a9b1a8570c8f/src/modules/ekf2/EKF2.cpp#L1561-L1570